### DISCUSSION TITLE
Project chrono Msys2 package

### DISCUSSION THREAD

**Discussion by DonBetur** [projectchrono/chrono#513] [Ideas]
This discussion is about making a msys2 package to perform easy installation for Project chrono. This may allow more user-friendly installation using pacman and auto-pull dependencies for different platforms, such as x86_64 etc.

**Comment by DonBetur** [comment]
I think main problem with using Project chrono for most users is collecting the packages you need for modules. That's why I'm creating this topic.

**Comment by rserban** [comment]
For several of the dependencies required by some Chrono modules we provide shell scripts that download, configure, build, and install the necessary libraries. See https://github.com/projectchrono/chrono/tree/main/contrib/build-scripts.

I am not familiar with msys2, but I have a few concerns about going down that route:
- we must support more than just Windows
- this would be yet another package that someone has to maintain and keep updated

Having said that, please feel free to submit a pull request. If this msys2 package is flexible enough (ideally, it should be able to build any and all of the Chrono modules but also allow a user to select which modules are of interest) we will certainly consider making it available to others.

**Reply by DonBetur** [reply]
Thanks for your answer! I gladly will check how it should be done. Msys2 packages are mostly for windows, but for different architectures. It's optional to do it for every single one, for example, as a start, msys packaging can be done for x86_64 platforms, which are most common for Windows users.

**Comment by DonBetur** [comment]
As I said such packages as:
Boost
Glm
Glfw 
Glew
OpenCascade 
Blaze
(This just what I remember)
Are up to date and frequently updating, so these dependecies can be used already.

**Comment by rserban** [comment]
Chrono does not use Boost.  
The ones you listed are just a few of the dependencies of certain Chrono modules.  

I'm looking forward to seeing how this shapes up. Hopefully, it's possible to satisfy the conditions I mentioned above.

Thanks

**Reply by DonBetur** [reply]
checked ProjectChrono installation guides, beg your pardon, boost needed for old blaze versions, I forgot about that.

**Comment by DonBetur** [comment]
Here, we have an update on this issue (hooray!)
Here, project chrono is added as a msys2 package:
https://github.com/msys2/MINGW-packages/pull/22466
This pull request is merged with msys2 packages master branch, so it's ready for use with
pacman -S projectchrono 
As well as projectchrono-docs package.

Although we can't add proprietary modules, this will allow more people to simply install project chrono with one simple command, we will add more modules in the future, now it has just core, irrlicht, python.

No CUDA or Intel MKL as I said above, sadly.
